api/envoy/api/v2/core/protocol.proto

```diff
@@ -44,6 +44,10 @@ message HttpProtocolOptions {
   // maximum number of request headers allowed is 100. Requests that exceed this limit will receive
   // a 431 response for HTTP/1.x and cause a stream reset for HTTP/2.
   google.protobuf.UInt32Value max_headers_count = 2 [(validate.rules).uint32 = {gte: 1}];
+
+  // If true, updating SNI automatically by reading host/authority on request header so that
+  // updating upstream cluster name
```

Comment:
Can you flesh this out a bit more in terms of what this actually does? Meaning, this will cause the connection pool to create new connections with this SNI set, etc. Can you also add a release note about this feature?